thriftbp/client_pool.go

```diff
@@ -259,6 +275,10 @@ func NewCustomClientPool(
 	protoFactory thrift.TProtocolFactory,
 	middlewares ...thrift.ClientMiddleware,
 ) (ClientPool, error) {
+	err := cfg.Validate()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
🔕 I do think this approach could lead to diverging behavior in the long term, though. e.g. there are some new restrictions added in the future, that then `newClientPool` would return an error because the config is bad, but we didn't add the same check into `Validate` so it still returns `nil`. I'm not sure what's the best solution for that, though.

Reply:
True, I've been trying to dig a bit deeper into this but in any case it's hard to keep track of this so it stays consistent.